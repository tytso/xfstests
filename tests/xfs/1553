#! /bin/bash
# SPDX-License-Identifier: GPL-2.0-or-later
# Copyright (c) 2022-2025 Oracle.  All Rights Reserved.
#
# FS QA Test No. 1553
#
# Populate a XFS filesystem and fuzz every metadir subdir field.
# Try online repair and, if necessary, offline repair,
# to test the most likely usage pattern.

. ./common/preamble
_begin_fstest dangerous_fuzzers scrub repair fuzzers_bothrepair

_register_cleanup "_cleanup" BUS

. ./common/filter
. ./common/populate
. ./common/fuzzy

_require_xfs_scratch_metadir
_require_scratch_xfs_fuzz_fields
_disable_dmesg_check

echo "Format and populate"
_scratch_populate_cached nofill > $seqres.full 2>&1

inode_ver=$(_scratch_xfs_get_metadata_field "core.version" 'path -m /rtgroups')

echo "Fuzz metadir subdir"
_scratch_xfs_fuzz_metadata '' 'both' 'path -m /rtgroups' >> $seqres.full
echo "Done fuzzing metadir subdir"

# success, all done
status=0
exit
