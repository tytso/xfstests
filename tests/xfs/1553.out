QA output created by 1553
Format and populate
Fuzz metadir subdir
core.mode = zeroes: offline re-scrub failed (1).
core.mode = firstbit: online repair failed (1).
core.mode = middlebit: online repair failed (1).
core.mode = lastbit: online repair failed (1).
core.mode = add: online repair failed (1).
core.metatype = zeroes: offline scrub didn't fail.
core.metatype = zeroes: online scrub didn't fail.
core.metatype = lastbit: offline scrub didn't fail.
core.metatype = lastbit: online scrub didn't fail.
core.uid = ones: online repair failed (1).
core.uid = firstbit: online repair failed (1).
core.uid = middlebit: online repair failed (1).
core.uid = lastbit: online repair failed (1).
core.uid = add: online repair failed (1).
core.uid = sub: online repair failed (1).
core.gid = ones: online repair failed (1).
core.gid = firstbit: online repair failed (1).
core.gid = middlebit: online repair failed (1).
core.gid = lastbit: online repair failed (1).
core.gid = add: online repair failed (1).
core.gid = sub: online repair failed (1).
core.nlinkv2 = zeroes: online repair failed (4).
core.projid_lo = ones: offline scrub didn't fail.
core.projid_lo = ones: online scrub didn't fail.
core.projid_lo = firstbit: offline scrub didn't fail.
core.projid_lo = firstbit: online scrub didn't fail.
core.projid_lo = middlebit: offline scrub didn't fail.
core.projid_lo = middlebit: online scrub didn't fail.
core.projid_lo = lastbit: offline scrub didn't fail.
core.projid_lo = lastbit: online scrub didn't fail.
core.projid_lo = add: offline scrub didn't fail.
core.projid_lo = add: online scrub didn't fail.
core.projid_lo = sub: offline scrub didn't fail.
core.projid_lo = sub: online scrub didn't fail.
core.projid_hi = ones: offline scrub didn't fail.
core.projid_hi = ones: online scrub didn't fail.
core.projid_hi = firstbit: offline scrub didn't fail.
core.projid_hi = firstbit: online scrub didn't fail.
core.projid_hi = middlebit: offline scrub didn't fail.
core.projid_hi = middlebit: online scrub didn't fail.
core.projid_hi = lastbit: offline scrub didn't fail.
core.projid_hi = lastbit: online scrub didn't fail.
core.projid_hi = add: offline scrub didn't fail.
core.projid_hi = add: online scrub didn't fail.
core.projid_hi = sub: offline scrub didn't fail.
core.projid_hi = sub: online scrub didn't fail.
core.forkoff = firstbit: online repair failed (1).
core.rtinherit = ones: offline scrub didn't fail.
core.rtinherit = ones: online scrub didn't fail.
core.rtinherit = firstbit: offline scrub didn't fail.
core.rtinherit = firstbit: online scrub didn't fail.
core.rtinherit = middlebit: offline scrub didn't fail.
core.rtinherit = middlebit: online scrub didn't fail.
core.rtinherit = lastbit: offline scrub didn't fail.
core.rtinherit = lastbit: online scrub didn't fail.
core.rtinherit = add: offline scrub didn't fail.
core.rtinherit = add: online scrub didn't fail.
core.rtinherit = sub: offline scrub didn't fail.
core.rtinherit = sub: online scrub didn't fail.
core.projinherit = ones: offline scrub didn't fail.
core.projinherit = ones: online scrub didn't fail.
core.projinherit = firstbit: offline scrub didn't fail.
core.projinherit = firstbit: online scrub didn't fail.
core.projinherit = middlebit: offline scrub didn't fail.
core.projinherit = middlebit: online scrub didn't fail.
core.projinherit = lastbit: offline scrub didn't fail.
core.projinherit = lastbit: online scrub didn't fail.
core.projinherit = add: offline scrub didn't fail.
core.projinherit = add: online scrub didn't fail.
core.projinherit = sub: offline scrub didn't fail.
core.projinherit = sub: online scrub didn't fail.
core.nosymlinks = zeroes: online repair failed (1).
core.nosymlinks = firstbit: online repair failed (1).
core.nosymlinks = middlebit: online repair failed (1).
core.nosymlinks = lastbit: online repair failed (1).
core.nosymlinks = add: online repair failed (1).
core.nosymlinks = sub: online repair failed (1).
next_unlinked = add: online scrub didn't fail.
next_unlinked = add: offline re-scrub failed (1).
next_unlinked = add: offline post-mod scrub failed (1).
v3.change_count = zeroes: offline scrub didn't fail.
v3.change_count = zeroes: online scrub didn't fail.
v3.change_count = ones: offline scrub didn't fail.
v3.change_count = ones: online scrub didn't fail.
v3.change_count = firstbit: offline scrub didn't fail.
v3.change_count = firstbit: online scrub didn't fail.
v3.change_count = middlebit: offline scrub didn't fail.
v3.change_count = middlebit: online scrub didn't fail.
v3.change_count = lastbit: offline scrub didn't fail.
v3.change_count = lastbit: online scrub didn't fail.
v3.change_count = add: offline scrub didn't fail.
v3.change_count = add: online scrub didn't fail.
v3.change_count = sub: offline scrub didn't fail.
v3.change_count = sub: online scrub didn't fail.
v3.flags2 = zeroes: online repair failed (1).
v3.flags2 = ones: online repair failed (1).
v3.flags2 = firstbit: online scrub didn't fail.
v3.flags2 = firstbit: offline re-scrub failed (1).
v3.flags2 = firstbit: offline post-mod scrub failed (1).
v3.flags2 = middlebit: online scrub didn't fail.
v3.flags2 = middlebit: offline re-scrub failed (1).
v3.flags2 = middlebit: offline post-mod scrub failed (1).
v3.flags2 = lastbit: online repair failed (1).
v3.flags2 = add: online repair failed (1).
v3.flags2 = sub: online repair failed (1).
v3.nrext64 = zeroes: offline scrub didn't fail.
v3.nrext64 = zeroes: online scrub didn't fail.
v3.nrext64 = firstbit: offline scrub didn't fail.
v3.nrext64 = firstbit: online scrub didn't fail.
v3.nrext64 = middlebit: offline scrub didn't fail.
v3.nrext64 = middlebit: online scrub didn't fail.
v3.nrext64 = lastbit: offline scrub didn't fail.
v3.nrext64 = lastbit: online scrub didn't fail.
v3.nrext64 = add: offline scrub didn't fail.
v3.nrext64 = add: online scrub didn't fail.
v3.nrext64 = sub: offline scrub didn't fail.
v3.nrext64 = sub: online scrub didn't fail.
v3.metadata = zeroes: online repair failed (1).
v3.metadata = firstbit: online repair failed (1).
v3.metadata = middlebit: online repair failed (1).
v3.metadata = lastbit: online repair failed (1).
v3.metadata = add: online repair failed (1).
v3.metadata = sub: online repair failed (1).
a.sfattr.list[0].parent = zeroes: online repair failed (1).
a.sfattr.list[0].parent = firstbit: online repair failed (1).
a.sfattr.list[0].parent = sub: online repair failed (1).
a.sfattr.list[0].parent_dir.inumber = lastbit: online repair failed (1).
a.sfattr.list[0].parent_dir.inumber = add: online repair failed (1).
a.sfattr.list[0].parent_dir.gen = zeroes: online repair failed (1).
a.sfattr.list[0].parent_dir.gen = ones: online repair failed (1).
a.sfattr.list[0].parent_dir.gen = firstbit: online repair failed (1).
a.sfattr.list[0].parent_dir.gen = middlebit: online repair failed (1).
a.sfattr.list[0].parent_dir.gen = lastbit: online repair failed (1).
a.sfattr.list[0].parent_dir.gen = add: online repair failed (1).
Done fuzzing metadir subdir
